source/uwp/Renderer/lib/AdaptiveCardRendererComponent.cpp

```diff
@@ -138,7 +134,7 @@ namespace AdaptiveNamespace
 
             if (m_explicitDimensions)
             {
-                RETURN_IF_FAILED(m_xamlBuilder->SetFixedDimensions(m_desiredWidth, m_desiredHeight));
+                m_xamlBuilder->SetFixedDimensions(m_desiredWidth, m_desiredHeight);
```

Comment:
>m_xamlBuilder->SetFixedDimensions(m_desiredWidth, m_desiredHeight); [](start = 16, length = 67)

I see that this method doesn't have any error cases, so changing the return to void is probably a perfectly reasonable thing to do, but i'll admit I feel slightly uncomfortable about it. I guess it's mostly about expecting to see consistent error checking on all methods at the callers (in a hypothetical world were we're consistently HRESULT based), and theoretically about the need to remember to update all callers if this ever gets error conditions in the future (admittedly unlikely, and internal so not really a huge problem). No need to change it necessarily, but I'd be interested if you have any particular thoughts on this, and/or if we have any coding conventions guidance.

Reply:
I don't see any explicit guidance in cppcoreguidelines, but at least in codebases I've been in, it's actually preferable to have a void return type if at all possible (obviously, it depends on what the function is actually *doing*). This alleviates the need for callers to anticipate a failure condition that may never happen. It feels a little weirder here since we're simply doing a `RETURN_IF_FAILED`, but there have been problems in the past with people writing error-handling code that never runs.

and then maybe someday it does run. That can be a real problem.

In code that exists in a tight contract (e.g. COM interface impls), it makes sense to pass `HRESULT`s if for no reason other than you might not have a chance to change in the future. With COM you also have no guarantee that code won't run between your function returning and the caller getting the return value (think proxy/stubs). However, when you control the caller and the callee and have no promises to keep with a contract, I think it's better to just rename the function if you need to change the return type. Going from "this function can never fail" to "this function might fail" feels like a pretty fundamental change in functionality.

---
In reply to: [325828953](https://github.com/microsoft/AdaptiveCards/pull/3457#discussion_r325828953) [](ancestors = 325828953)